### PR TITLE
Add mac address validation for BootMACAddress

### PR DESCRIPTION
**Pull request by ardaguclu**
This PR adds mac address validation for BootMACAddress if it is not empty.

### CODE REVIEW

File: apis/metal3.io/v1alpha1/baremetalhost_validation_test.go [hunk 1]
```diff
@@ -263,6 +263,22 @@ func TestValidateCreate(t *testing.T) {
...
+				Spec: BareMetalHostSpec{
+					BMC: BMCDetails{
+						Address:         "irmc://127.0.1.1",
+						CredentialsName: "test1",
+					},
+					BootMACAddress: "00:00:00:00:00",
+					BootMode:       UEFISecureBoot,
+				}},
+			oldBMH:    nil,
+			wantedErr: "address 00:00:00:00:00: invalid MAC address",
```
Comment (furkatgofurov7) [inline apis/metal3.io/v1alpha1/baremetalhost_validation_test.go:280]: is it missing one more `00` at the end?
```suggestion
			wantedErr: "address 00:00:00:00:00:00: invalid MAC address",
```

Comment (dtantsur) [reply apis/metal3.io/v1alpha1/baremetalhost_validation_test.go:280]: I think it was done specifically to make it invalid?

Comment (ardaguclu) [reply apis/metal3.io/v1alpha1/baremetalhost_validation_test.go:280]: Yes, it is deliberately incorrect for the test

Comment (furkatgofurov7) [reply apis/metal3.io/v1alpha1/baremetalhost_validation_test.go:280]: my bad disregard, I overlooked BootMACAddress on line 276

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Comment by furkatgofurov7** [comment]
> approve

Looks like prow was not able to interpret this for some unknown reason, @dtantsur can you please try again?

**Comment by zhouhao3** [comment]
lgtm

**Review by furkatgofurov7 — APPROVED** [review]

**Pull request merged**